src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -25,37 +26,44 @@
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
 
-    private final Index targetIndex;
+    private ArrayList<Index> targetIndexArraylist;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(ArrayList<Index> targetIndexArraylist) {
+        this.targetIndexArraylist = targetIndexArraylist;
     }
 
 
     @Override
     public CommandResult executeUndoableCommand() throws CommandException {
 
         List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
+        String result = "";
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
-        }
+        for (Index i : targetIndexArraylist) {
+            if (i.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+
+            ReadOnlyPerson personToDelete = lastShownList.get(i.getZeroBased());
+
+            try {
+                model.deletePerson(personToDelete);
+            } catch (PersonNotFoundException pnfe) {
+                assert false : "The target person cannot be missing";
+            }
+
+            result += String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete) + "\n";
 
```

Comment:
Do we want to display the `MESSAGE_DELETE_PERSON_SUCCESS` n times? Was wondering if we can make this more succinct, perhaps by showing 

`MESSAGE_DELETE_PERSON_SUCCESS`:
[Person 1]
[Person 2]
...